src/auth.test.js

```diff
@@ -106,5 +111,37 @@ describe('auth', () => {
       channel.close()
       expect(unsubscribe.mock.calls.length).toBe(1)
     })
+
+    it('emits user or error', () => {
+      const userMock = {uid: 'uid'}
+      const errorMock = {code: 'auth/error-code'}
+      const emit = (user, authError) => {
+        subs.forEach(({ nextOrObserver, error }) => {
+          if (user) {
+            nextOrObserver(user)
+          }
+          if (authError) {
+            error(authError)
+          }
+        })
+      }
+      const channel = authModule.channel.call({
+        ...context,
+        // reset memoized channel
+        _authChannel: null
+      })
+
+      const userSpy = ({ user }) => {
+        expect(user).toEqual(userMock)
+      }
+      channel.take(userSpy)
+      emit(userMock, null)
+
+      const errorSpy = ({ error }) => {
+        expect(error).toEqual(errorMock)
```

Comment:
How can we make sure that those lines are executed?
Could we make the test otherwise?